src/bin/bot_pid_control/app.cpp

```diff
@@ -413,3 +419,81 @@ void jaiabot::apps::BotPidControl::handle_command(const jaiabot::protobuf::PIDCo
     }
 
 }
+
+// Handle DesiredSetpoint messages from high_control.proto
+
+void jaiabot::apps::BotPidControl::handle_command(const jaiabot::protobuf::DesiredSetpoints& command)
+{
+    glog.is_debug1() && glog << "Received command: " << command.ShortDebugString() << std::endl;
+
+    lastCommandReceived = goby::time::SystemClock::now<goby::time::MicroTime>();
+    
+    switch (command.type()) {
+        case jaiabot::protobuf::SETPOINT_STOP:
+            throttle = 0.0;
+            throttleMode = MANUAL;
+            break;
+        case jaiabot::protobuf::SETPOINT_IVP_HELM:
+            handle_helm_course(command.helm_course());
+            break;
+        case jaiabot::protobuf::SETPOINT_REMOTE_CONTROL:
+            handle_remote_control(command.remote_control());
+            break;
+        case jaiabot::protobuf::SETPOINT_DIVE:
+            handle_dive_depth(command.dive_depth());
+            break;
+        case jaiabot::protobuf::SETPOINT_POWERED_ASCENT:
+            handle_powered_ascent();
+            break;
+    }
+}
+
+void jaiabot::apps::BotPidControl::handle_helm_course(const goby::middleware::frontseat::protobuf::DesiredCourse& desired_course) {
+
+    if (desired_course.has_heading()) {
+        rudder_is_using_pid = true;
+        target_heading = desired_course.heading();
+    }
+    if (desired_course.has_speed()) {
+        throttleMode = PID_SPEED;
+        target_speed = desired_course.speed();
+    }
+    // TO DO:  PID for the depth that uses elevators while moving forward
```

Comment:
We can ignore pitch, roll, z_rate, and altitude options for now. The IvP behaviors I'm using only use the heading, speed, and depth control domains.